docs(material/sidenav): Deprecated methods used in the responsive sidenav example

lsamboretrorabbit opened the pull request:
Fixes the bug in the documentation - Angular Material 'sidenav' component. 
Updated the file 'sidenav-responsive-example' to use the CDK instead of using media queries

Fixes #29266

lsamboretrorabbit committed:
docs(material/sidenav): Deprecated methods used in the responsive sidenav example

Fixes the bug in the documentation - Angular Material 'sidenav' component.
Updated the ts file to use the CDK instead of using media queries

Fixes #29266

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.ts:
```diff
@@ -29,16 +35,23 @@ export class SidenavResponsiveExample implements OnDestroy {
-  private _mobileQueryListener: () => void;
+  mobileQuery: boolean;
+  private breakpointSubscription: Subscription;
 
```
It looks like this file has some lint failures.

lsamboretrorabbit committed:
removed unused

lsamboretrorabbit committed:
update private variables

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.html:
```diff
@@ -1,25 +1,21 @@
 @if (shouldRun) {
-  <div class="example-container" [class.example-is-mobile]="mobileQuery.matches">
+  <div class="example-container" [class.example-is-mobile]="mobileQuery">
```
We can clean this up even more with the `async` pipe instead of having to subscribe manually. Here's an example: https://github.com/angular/components/blob/main/src/dev-app/screen-type/screen-type-demo.html#L4

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.html:
```diff
@@ -1,25 +1,21 @@
+  <div class="example-container" [class.example-is-mobile]="mobileQuery">
     <mat-toolbar color="primary" class="example-toolbar">
-      <button mat-icon-button (click)="snav.toggle()"><mat-icon>menu</mat-icon></button>
+      <button mat-icon-button (click)="snav.toggle()">
```
`snav` -> `sidenav`

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.html:
```diff
@@ -1,25 +1,21 @@
-          @for (nav of fillerNav; track nav) {
-            <a mat-list-item routerLink=".">{{nav}}</a>
-          }
+          <a mat-list-item *ngFor="let nav of fillerNav">{{ nav }}</a>
```
Why was this switched to `ngFor`? The whole repo intentionally uses the new control flow syntax.

lsamboretrorabbit committed:
docs(material/sidenav): Deprecated methods used in the responsive sidenav example

Fixes the bug in the documentation - Angular Material 'sidenav' component.
Updated the file 'sidenav-responsive-example' to use the CDK instead of using media queries

Fixes #29266

lsamboretrorabbit commented:
@crisbeto thank you, i have made updates

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.ts:
```diff
@@ -1,24 +1,31 @@
   standalone: true,
-  imports: [MatToolbarModule, MatButtonModule, MatIconModule, MatSidenavModule, MatListModule],
+  imports: [
+    CommonModule,
```
Import the `AsyncPipe` here directly instead of the `CommonModule`.

crisbeto reviewed (commented):
Looks good, aside from one last comment and resolving the merge conflicts.

lsamboretrorabbit committed:
docs(material/sidenav): Deprecated methods used in the responsive sidenav example

Fixes the bug in the documentation - Angular Material 'sidenav' component.
Updated the file 'sidenav-responsive-example' to use the CDK instead of using media queries

Fixes #29266

lsamboretrorabbit committed:
Merge branch 'main' into fix-docs-sidenav-example

crisbeto commented on src/components-examples/material/sidenav/sidenav-responsive/sidenav-responsive-example.ts:
```diff
@@ -29,16 +36,27 @@ export class SidenavResponsiveExample implements OnDestroy {
   );
 
-  private _mobileQueryListener: () => void;
+  mobileQuery: boolean;
```
I think that this and `_breakpointSubscription` aren't used anymore and can be removed.

crisbeto reviewed (commented):
I see that there's a merge commit in the PR now. The tooling that we use to merge PRs doesn't allow merge commits. Can you drop it from your branch?

lsamboretrorabbit committed:
Revert "Merge branch 'main' into fix-docs-sidenav-example"

This reverts commit 6234e9fc6f7908121c348b10006dee61cec6cbea, reversing
changes made to 4a1169759b3aff30cabdbfa4c7ffeb851ef6c0dd.

lsamboretrorabbit committed:
docs(material/sidenav): Deprecated methods used in the responsive sidenav example

Fixes the bug in the documentation - Angular Material 'sidenav' component.
Updated the file 'sidenav-responsive-example' to use the CDK instead of using media queries

Fixes #29266

lsamboretrorabbit committed:
Merge branch 'main' into fix-docs-sidenav-example
